apps/site/pages/en/learn/modules/package-configuration.mdx

```diff
@@ -0,0 +1,597 @@
+
+This is almost exactly the same as the CJS-CJS configuration above with 1 small difference: the [`"type"`](https://nodejs.org/api/packages.html#type) field.
+
+**Working example**: [esm-with-esm-distro](https://github.com/JakobJingleheimer/nodejs-module-config-examples/tree/main/packages/esm/esm-distro)
+
+```json displayName="Minimal package.json"
+{
+  "name": "esm-source-and-distribution",
+  "type": "module"
+  // "main": "./index.js"
+}
+```
+
+```json displayName="Advanced (verbose) package.json"
+{
+  "name": "esm-source-and-distribution",
+  "type": "module",
+  "engines": { "node": ">=12.22.7" },
+  "exports": {
+    ".": "./dist/index.js",
+    "./package.json": "./package.json"
+  }
+}
+```
+
+Note that ESM now _is_ “backwards” compatible with CJS: a CJS module now _can_ [`require()` an ES Module](https://nodejs.org/api/modules.html#loading-ecmascript-modules-using-require) without a flag as of 23.0.0 and 22.12.0.
+
+### CJS source and only ESM distribution
+
+This takes a small bit of finesse but is also pretty straight-forward.
+
+**Working example**: [cjs-with-esm-distro](https://github.com/JakobJingleheimer/nodejs-module-config-examples/tree/main/packages/cjs/esm-distro)
+
+```json displayName="Minimal package.json"
+{
+  "name": "cjs-source-with-esm-distribution",
+  "main": "./dist/index.mjs"
+}
+```
+
+```json displayName="Advanced (verbose) package.json"
+{
+  "name": "cjs-source-with-esm-distribution",
+  "type": "commonjs", // current default, but may change
+  "engines": { "node": ">=12.22.7" },
+  "exports": {
+    ".": "./dist/index.mjs",
+    "./package.json": "./package.json"
+  }
+}
+```
+
+The [`.mjs`](https://nodejs.org/api/esm.html#enabling) file extension is a trump-card: it will override **any** other configuration and the file will be treated as ESM. Using this file extension is necessary because `packageJson.exports.import` does **NOT** signify that the file is ESM (contrary to common, if not universal, misperception), only that it is the file to be used when the package is imported (ESM _can_ import CJS. See [Gotchas](#gotchas) below).
+
+### CJS source and both CJS & ESM distribution
+
+You have a few options:
+
+#### Attach named exports directly onto `exports`
+
+Classic but takes some sophistication and finesse.
+
+Pros:
+
+- Smaller package weight
+- Easy and simple (probably least effort if you don't mind keeping to a minor syntax stipulation)
+- Precludes [the Dual-Package Hazard](#the-dual-package-hazard)
+
+Cons:
+
+- Requires very specific syntax (either in source code and/or bundler gymnastics).
+
+**Working example**: [cjs-with-dual-distro (properties)](https://github.com/JakobJingleheimer/nodejs-module-config-examples/tree/main/packages/cjs/dual/property-distro)
+
+```json displayName="Minimal package.json"
```

Comment:
I'd move example first so the reader understand what we are talking about and then pros and cons, also I'd put some explanation at the beginning of the paragraph on what `Attach named exports directly onto `exports`` means